core/domain/blog_statistics_domain.py

```diff
@@ -0,0 +1,1165 @@
+    """Returns a dict with number of views month wise for the ongoing year.
+
+    Args:
+        stats: BlogPostViewsAggregatedStats|AuthorBlogPostViewsAggregatedStats.
+            The stats domain object for which yearly views are to be generated.
+
+    Returns:
+        dict. A dict with number of views keyed to months in UTC fromat in the
+        ongoing year.
+    """
+    current_year = datetime.datetime.utcnow().strftime('%Y')
+    # We add '_' infront of month keys to avoid auto ordering of keys
+    # in frontend, otherwise the keys get treated like numeric keys instead of
+    # strings by typescript.
+    stats_dict = {
+        '_' + k: stats.views_by_month[current_year][k] for k in list(
+            stats.views_by_month[current_year])
+    }
+    return stats_dict
+
+
+@overload
+def _generate_past_twenty_four_hour_reads_stats_from_reads_by_hour(
+    stats: BlogPostReadsAggregatedStats,
+) -> Dict[str, int]: ...
+
+
+@overload
+def _generate_past_twenty_four_hour_reads_stats_from_reads_by_hour(
+    stats: AuthorBlogPostReadsAggregatedStats,
+) -> Dict[str, int]: ...
+
+
+def _generate_past_twenty_four_hour_reads_stats_from_reads_by_hour(
+    stats: Union[
+        BlogPostReadsAggregatedStats,
+        AuthorBlogPostReadsAggregatedStats
+    ]
+) -> Dict[str, int]:
+    """Returns a dict with number of reads hour wise for the past twenty four
+    hours.
+
+    Args:
+        stats: BlogPostReadsAggregatedStats|AuthorBlogPostReadsAggregatedStats.
+            The stats domain object for which hourly reads are to be generated.
+
+    Returns:
+        dict. A dict with number of reads keyed to hour('HH') in UTC fromat for
+        the past twenty four hours.
+    """
+    current_hour = datetime.datetime.utcnow().hour
+    current_date = datetime.datetime.utcnow().strftime('%Y-%m-%d')
+    yesterdays_date = (
+        datetime.datetime.utcnow() - datetime.timedelta(days=1)
+    ).strftime('%Y-%m-%d')
+    deficit_hours_in_current_date = 24 - current_hour - 1
+    hourly_stats = {}
+    if yesterdays_date in stats.reads_by_hour.keys():
+        yesterday_stats = stats.reads_by_hour[yesterdays_date]
+        # We add '_' infront of hour key to avoid auto ordering of keys in
+        # frontend, otherwise the keys get treated like numeric keys instead of
+        # strings by typescript.
+        hourly_stats = {
+            '_' + k: yesterday_stats[k] for k in list(
+                yesterday_stats)[- deficit_hours_in_current_date:]
+        }
+    todays_stats = stats.reads_by_hour[current_date]
+    hourly_stats.update({
+        '_' + k: todays_stats[k] for k in list(todays_stats)[:current_hour + 1]
+    })
+    return hourly_stats
+
+
+@overload
+def _generate_past_week_reads_stats_from_reads_by_date(
+    stats: BlogPostReadsAggregatedStats,
+) -> Dict[str, int]: ...
+
+
+@overload
+def _generate_past_week_reads_stats_from_reads_by_date(
+    stats: AuthorBlogPostReadsAggregatedStats,
+) -> Dict[str, int]: ...
+
```

Comment:
Please share MyPy error if method is defined without overload!

Reply:
No error is thrown after changing return type to `none`!